vulture/config.py

```diff
@@ -0,0 +1,214 @@
+"""
+This module handles retrieval of configuration values from either the
+command-line arguments or the pyproject.toml file.
+"""
+import argparse
+import sys
+from os.path import abspath, exists
+
+import toml
+
+from .version import __version__
+
+#: Possible configuration options and their respective defaults
+DEFAULTS = {
+    "min_confidence": 0,
+    "paths": [],
+    "exclude": [],
+    "ignore_decorators": [],
+    "ignore_names": [],
+    "make_whitelist": False,
+    "sort_by_size": False,
+    "verbose": False,
+}
+
+#: sentinel value to distinguish between "False" and "no default given"
+NO_DEFAULT = object()
+
+
+def _check_config_types(data):
+    """
+    Checks the types of the values in *data* against the expected types of
+    config-values. If a value is of the wrong type it will raise a SystemExit.
+    """
+    given_types = {
+        (key, type(value))
+        for key, value in data.items()
+        if key in DEFAULTS and value is not NO_DEFAULT
+    }
+    expected_types = {(key, type(value)) for key, value in DEFAULTS.items()}
+    incorrect_types = given_types - expected_types
+    if incorrect_types:
+        keys = ", ".join(sorted(item[0] for item in incorrect_types))
+        msg = f"Incorrect data-types for the config values: {keys}"
+        sys.exit(msg)
```

Comment:
Let's rename this function to _check_input_config and combine the key and type checking: 

``` python
for key, value in data.items():
    if key not in DEFAULTS:
        sys.exit(f"Unknown configuration key: {key}")
    if value is not NO_DEFAULT and type(value) != type(DEFAULTS[key]):
        sys.exit(f"Data type for {key} must be {DEFAULTS[key]}")
```

Reply:
Done in 56c4f66

I still kept the print to stderr because I prefer those kinds of messages on stderr in case stdout is redirected. May happen in CI pipelines.